op-scim-bridge/templates/secret.yaml

```diff
@@ -1,7 +1,9 @@
-{{- if .Values.scim.credentialsSecret }}
-{{- if or .Values.scim.credentialsSecret.value_json .Values.scim.credentialsSecret.value_base64 }}
-{{- if and .Values.scim.credentialsSecret.value_json .Values.scim.credentialsSecret.value_base64 }}
-  {{- fail "Only one of scim.credentialsSecret.value_json and scim.credentialsSecret.value_base64 can be specified" }}
+{{- if .Values.scim.credentialsSecret.enabled }}
+{{- if and .Values.scim.credentialsSecret.enabled .Values.scim.credentialsVolume.enabled }}
+  {{- fail "Either credentialsSecret or credentialsVolume may be specified, not both"}}
+{{- end }}
+{{- if not .Values.scim.credentialsSecret.data }}
+  {{- fail "Contents of the scimsession file must be provided" }}
```

Comment:
Given we want to keep both data options, this will probably have to look something like:

```yaml
{{- if and (.Values.scim.credentialsSecret.data) (.Values.scim.credentialsSecret.data_base64) -}}
  {{- fail "Only one of scim.credentialsSecret.data and cim.credentialsSecret.data_base64 can be specified" -}}
{{- end -}}

Reply:
I would recommend this be brought out of this logic loop (maybe a helper template function) since users of helm would expect it to function a certain way, which is counter to how this works. I can understand the need to keep backward compatibility, but you might be able to do both by moving it.